casper/src/main/scala/io/casperlabs/casper/highway/MessageExecutor.scala

```diff
@@ -96,47 +91,51 @@ class MessageExecutor[F[_]: Concurrent: Log: Time: Metrics: BlockStorage: DagSto
       _ <- markDeploysAsProcessed(message)
             .timer("markDeploysAsProcessed")
             .whenA(message.isBlock)
+      _ <- MultiParentFinalizer[F].addMessage(message)
       // Forking event emissions so as not to hold up block processing.
       w1 <- BlockEventEmitter[F].blockAdded(message.messageHash).timer("emitBlockAdded").forkAndLog
-      w2 <- updateLastFinalizedBlock(message).timer("updateLastFinalizedBlock")
-    } yield w1 *> w2
+    } yield w1
 
-  private def updateLastFinalizedBlock(message: Message): F[F[Unit]] =
-    for {
-      results <- MultiParentFinalizer[F].onNewMessageAdded(message)
-      w <- results.toList
-            .traverse {
-              case MultiParentFinalizer.FinalizedBlocks(newLFB, _, finalized, orphaned) => {
-                val lfbStr = newLFB.show
-                val finalizedStr = finalized
-                  .filter(_.isBlock)
-                  .map(_.messageHash)
-                  .map(PrettyPrinter.buildString)
-                  .mkString("{", ", ", "}")
-                for {
-                  _ <- Log[F].info(
-                        s"New last finalized block hashes are ${lfbStr -> null}, ${finalizedStr -> null}. Orphaned ${orphaned.size} messages."
-                      )
-                  finalizedBlockHashes = finalized.filter(_.isBlock).map(_.messageHash)
-                  _ <- DeployBuffer[F]
-                        .removeFinalizedDeploys(finalizedBlockHashes + newLFB)
-                  // Ballots cannot be really finalized but we mark them as such in the DAG
-                  // to improve the performance of the finalizer (that has to follow all justifications).
-                  // Send out notification about blocks ONLY.
-                  orphanedBlockHashes = orphaned.filter(_.isBlock).map(_.messageHash)
-                  _ <- BlockEventEmitter[F]
-                        .newLastFinalizedBlock(
-                          newLFB,
-                          finalizedBlockHashes,
-                          orphanedBlockHashes
+  def checkFinality: F[F[Unit]] =
```

Comment:
```suggestion
  def checkFinality(): F[F[Unit]] =
```
Sorry, missed this earlier.

Reply:
Me too.

Reply:
Done.